src/Features/CSharp/Portable/ConvertSwitchStatementToExpression/ConvertSwitchStatementToExpressionCodeFixProvider.Rewriter.cs

```diff
@@ -0,0 +1,263 @@
+                var generateDeclaration = rewriter.TryRemoveVariableDeclarators(node, semanticModel, editor);
+                var finalStatement = rewriter.GetFinalStatement(switchExpression, generateDeclaration);
+                rewriter.Free();
+                return finalStatement;
+            }
+
+            private bool TryRemoveVariableDeclarators(SwitchStatementSyntax node, SemanticModel semanticModel, SyntaxEditor editor)
+            {
+                // Try to remove the varaiable declarator only if these are simple identifiers.
+                if (_assignmentTargets.Count == 0 ||
+                    !_assignmentTargets.All(target => target.IsKind(SyntaxKind.IdentifierName)))
+                {
+                    return false;
+                }
+
+                var symbols = _assignmentTargets.Select(target => semanticModel.GetSymbolInfo(target).Symbol);
+                // If all variables are local and data does not flows in for any of them, 
+                // we can assign the switch expression directly to a variable declaration.
+                foreach (var symbol in symbols)
+                {
+                    if (symbol == null)
+                    {
+                        return false;
+                    }
+
+                    if (symbol.Kind != SymbolKind.Local)
+                    {
+                        return false;
+                    }
+
+                    var syntaxReferences = symbol.DeclaringSyntaxReferences;
+                    if (syntaxReferences.Length != 1)
+                    {
+                        return false;
+                    }
+
+                    var declarator = (VariableDeclaratorSyntax)syntaxReferences[0].GetSyntax();
+                    if (declarator.Initializer != null)
+                    {
+                        return false;
+                    }
+
+                    var dataFlow = semanticModel.AnalyzeDataFlow(
+                        declarator.GetAncestor<StatementSyntax>(),
+                        node.GetPreviousStatement());
+
+                    if (!dataFlow.Succeeded)
+                    {
+                        return false;
+                    }
+
+                    if (dataFlow.ReadInside.Contains(symbol) ||
+                        dataFlow.WrittenInside.Contains(symbol))
+                    {
+                        return false;
+                    }
+                }
+
+                foreach (var symbol in symbols)
+                {
+                    editor.RemoveNode(symbol.DeclaringSyntaxReferences[0].GetSyntax());
+                }
+
+                return true;
+            }
+
+            private StatementSyntax GetFinalStatement(ExpressionSyntax switchExpression, bool generateDeclaration)
+            {
+                if (_assignmentTargets.Count == 0)
+                {
+                    return ReturnStatement(switchExpression);
```

Comment:
if everything in the original switch was a throw... what do you generate?  it looks like it will be a standalone switch expression.  Is that legal in the language?

Reply:
Now we generate throw `e switch { .. }` for that.